tests/plotting/test_plot_manager.py

```diff
@@ -76,10 +76,64 @@ def drop(self, path: Path):
         del self.plots[self.plots.index(self.path / path.name)]
 
 
+class PlotRefreshTester:
+    plot_manager: PlotManager
+    expected_result: PlotRefreshResult
+    expected_result_matched: bool
+
+    def __init__(self, root_path: Path):
+        self.plot_manager = PlotManager(root_path, self.refresh_callback)
+        self.expected_result = PlotRefreshResult()
+        self.expected_result_matched = False
+
+    def refresh_callback(self, event: PlotRefreshEvents, refresh_result: PlotRefreshResult):
+        if event != PlotRefreshEvents.done:
+            # Only validate the final results for this tests
+            return
+        for name in ["loaded", "removed", "processed", "remaining"]:
+            try:
+                actual_value = refresh_result.__getattribute__(name)
+                if type(actual_value) == list:
+                    expected_list = self.expected_result.__getattribute__(name)
+                    if len(expected_list) != len(actual_value):
+                        return
+                    values_found = 0
+                    for value in actual_value:
+                        if type(value) == PlotInfo:
+                            for plot_info in expected_list:
+                                if plot_info.prover.get_filename() == value.prover.get_filename():
+                                    values_found += 1
+                                    continue
+                        else:
+                            if value in expected_list:
+                                values_found += 1
+                                continue
+                    if values_found != len(expected_list):
+                        log.error(f"{name} invalid: values_found {values_found} expected {len(expected_list)}")
+                        return
+                else:
+                    expected_value = self.expected_result.__getattribute__(name)
+                    if actual_value != expected_value:
+                        log.error(f"{name} invalid: actual {actual_value} expected {expected_value}")
+                        return
+            except AttributeError as error:
+                log.error(f"{error}")
+                return
+        self.expected_result_matched = True
```

Comment:
if you move this in, right before `except`, you don't need the `return` statement here either

Reply:
Changed